addons/point_of_sale/models/pos_order.py

```diff
@@ -696,49 +647,53 @@ def action_pos_order_paid(self):
 
     @api.multi
     def action_pos_order_invoice(self):
-        Invoice = self.env['account.invoice']
+        moves = self.env['account.move']
 
         for order in self:
             # Force company for all SUPERUSER_ID action
-            local_context = dict(self.env.context, force_company=order.company_id.id, company_id=order.company_id.id)
-            if order.invoice_id:
-                Invoice += order.invoice_id
+            if order.account_move:
+                moves += order.account_move
                 continue
 
             if not order.partner_id:
                 raise UserError(_('Please provide a partner for the sale.'))
 
-            invoice = Invoice.new(order._prepare_invoice())
-            invoice._onchange_partner_id()
-            invoice.fiscal_position_id = order.fiscal_position_id
-
-            inv = invoice._convert_to_write({name: invoice[name] for name in invoice._cache})
-            new_invoice = Invoice.with_context(local_context).sudo().create(inv)
+            move_vals = {
```

Comment:
Please keep function `_prepare_invoice`  or rename to `_prepare_account_move ` for purpose of inheritance on other modules

Reply:
@celm1990  I would suggest you to add these comments on https://github.com/odoo/odoo/pull/34766 as its seems for the Fixes/Imp

Reply:
We are already taking care of that, it will be merged soon #34554

Reply:
@celm1990 what do you mean?